package.json

```diff
@@ -5,6 +5,7 @@
   "description": "This JavaScript library provides a simple API to submit and retrieve user data to the Subspace network.",
   "main": "./dist/subspace.js",
   "types": "./dist/subspace.d.ts",
+  "type": "module",
```

Comment:
This is A HUGE PAIN because it will force people to convert their whole project to ESM or else they can't use it.

One of the authors of popular packages went pure ESM route for their packages and had to write this: https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c

From the fact that author had to create that article and number of comments under it you can guess how easy it is to use now (I used older version of `p-retry` just to avoid changing the whole project and reconfigure tooling around it.

So far I'm very strongly against ESM unless things get MUCH better.

Reply:
The reason of using EMS was related to some issues loading and injecting the polkadot `extension` in the `window` object, looks like there was a tiny delay between `extension-inject` loading the extension and `extension-dapp` requesting the injection status (win.injectedWeb3)

There is an optional parameter in the `web3Enable` method, added an array of functions returning a promise, the promise will run before the checks on `isWeb3Injected` adding some milliseconds to await the extension to be injected. 

Another option is checking the  `web3Enable` output and if there is no extensions loaded run a retry, this adds extra code and i decided to use the the existing optional parameter from `web3Enable` call.

https://github.com/subspace/subspace.js/blob/4d3acfd2cb0dc7e82039cd25812182bb74896dfa/src/identity.ts#L32-L37

Reply:
Hm, I'd propably reach out to Jaco, there must be a better way than timeouts. If ESM is absolutely required we can do what polkadot.js does and have both ESM and CommonJS version of the library using `exports` field in `package.json`.

Reply:
I will, it is not closed for me. I created a new issue to not forget.